src/_pytest/nodes.py

```diff
@@ -90,37 +91,27 @@ class Node(metaclass=NodeMeta):
     def __init__(
         self,
         name,
-        parent: Optional["Node"] = None,
-        config: Optional[Config] = None,
-        session: Optional["Session"] = None,
+        parent: Union["Node", "Session"],
         fspath: Optional[py.path.local] = None,
         nodeid: Optional[str] = None,
```

Comment:
Should we still allow to customize the entire node id? 

I know there are items which are derived from other items, for example flake8 items which are derived from normal Python items, but those can be customized by passing a different `name` already.

Reply:
actualy an oversight, we should also disallow it quickly for from_parent

Reply:
my  main intend with this pr was the structural details, so nodeid was not instrumental to  the data structure

Reply:
I see.

So the main intent is to remove, in 6.1, `config`, `session` and `nodeid` from the `Node` constructor, as I understand they are the cause of structural problems, correct?

Reply:
Yup